apollo-router/src/plugins/telemetry/fmt_layer.rs

```diff
@@ -167,28 +167,58 @@ where
         }
 
         BUF.with(|buf| {
-            let borrow = buf.try_borrow_mut();
-            let mut a;
-            let mut b;
-            let mut buf = match borrow {
-                Ok(buf) => {
-                    a = buf;
-                    &mut *a
-                }
-                _ => {
-                    b = String::new();
-                    &mut b
-                }
-            };
-            if self.fmt_event.format_event(&ctx, &mut buf, event).is_ok() {
-                let mut writer = self.make_writer.make_writer();
-                if let Err(err) = std::io::Write::write_all(&mut writer, buf.as_bytes()) {
-                    if err.kind() != std::io::ErrorKind::WouldBlock {
-                        let _ = std::io::stderr().write_all(
-                            format!("cannot flush the logging buffer, this is a bug: {err:?}")
-                                .as_bytes(),
-                        );
+            let mut buf = &mut *(buf.borrow_mut());
```

Comment:
Diff sims to contain unrelated change.
Can you fix this PR?

Reply:
Again, not sure how this got here. Rebased!